### DISCUSSION TITLE
Updates for msw are not correct

### DISCUSSION THREAD

**Discussion by Belco90** [renovatebot/renovate#18982] [Request Help] [answered]
Hi there 👋 

I'm using [msw](https://mswjs.io/) in my project, where renovate is set too. When renovate create PRs to update msw, it updates the dependency correctly, but it doesn't update the corresponding `mockServiceWorker.js` file needed to mock the endpoints in the browser.

This file is needed to register a Service Worker in the client so msw can intercept the requests. Msw updates this file when installing dependencies, so just running `pnpm install` after the updating it would be enought to update the `mockServiceWorker.js` file.

Would be possible to run this on renovate when updating msw, so the `mockServiceWorker.js` gets updated as expected.

**Comment by rarkins** [comment, marked as answer]
You can if you're self hosting using postUpgradeTasks: https://docs.renovatebot.com/configuration-options/#postupgradetasks

**Reply by Belco90** [reply]
I'm using the GitHub App. Any way of using the post upgrade task or another workaround there?

**Reply by HonkingGoose** [reply]
Hi @Belco90! 👋 

`postUpgradeTasks` only work when you're self-hosting Renovate. So you're out-of-luck with the GitHub app version of Renovate. 😞 

Here's the quote from our docs: [^docs]

> **Note**
>
> Post-upgrade tasks can only be used on self-hosted Renovate instances.

I don't know if we want/should change the Renovate code so it upgrades the MSW `mockServiceWorker.js` file. That's for @rarkins to decide. 😉 

Maybe you can use custom regex to update the `mockServiceWorker.js` file and the MSW package at the same time? See the [`regexManagers` docs](https://docs.renovatebot.com/configuration-options/#regexmanagers).

[^docs]: https://docs.renovatebot.com/configuration-options/#postupgradetasks

**Reply by Belco90** [reply]
Thanks @HonkingGoose!

> Maybe you can use custom regex to update the mockServiceWorker.js file and the MSW package at the same time? See the [regexManagers docs](https://docs.renovatebot.com/configuration-options/#regexmanagers).

This would be amazing, to be honest. The command run should be just installing dependencies (`npm/yarn/pnpm install`) so MSW itself takes care of updating the `mockServiceWorker.js` in a post-install task.

**Reply by rarkins** [reply]
This could be made possible only once someone implements post upgrade tasks being run in a sidecar container

**Reply by HonkingGoose** [reply]
Looks like we have two issues that cover `postUpgradeTasks` running in a sidecar container:

- #11018
- #8111
